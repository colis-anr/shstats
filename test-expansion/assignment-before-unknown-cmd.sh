#!/bin/dash

x=old
x=new /bin/test "whatever"
# we know statically that $x is "old" since the previous line is a process
echo $x
