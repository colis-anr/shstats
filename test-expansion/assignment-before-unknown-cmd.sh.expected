[
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdPrefix",
                [
                  "CmdPrefix_AssignmentWord",
                  [ [ "Name", "x" ], [ "Word", "old", [] ] ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdPrefix_CmdWord_CmdSuffix",
                [
                  "CmdPrefix_AssignmentWord",
                  [ [ "Name", "x" ], [ "Word", "new", [] ] ]
                ],
                [ "CmdWord_Word", [ "Word", "/bin/test", [] ] ],
                [ "CmdSuffix_Word", [ "Word", "\"whatever\"", [] ] ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName_CmdSuffix",
                [
                  "CmdName_Word",
                  [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                ],
                [ "CmdSuffix_Word", [ "Word", "old", [] ] ]
              ]
            ]
          ]
        ]
      ]
    ]
  ]
]